packages/lesswrong/server/editor/utils.ts

```diff
@@ -93,3 +95,48 @@ export const preProcessLatex = async (content) => {
 
   return content;
 }
+
+const revisionFieldsToCopy: (keyof DbRevision)[] = [
+  "originalContents",
+  "updateType",
+  "commitMessage",
+  "html",
+  "version",
+  "userId",
+  "editedAt",
+  "wordCount",
+];
+
+/**
+ * Make an editable document reflect the latest revision available
+ *
+ * Documents can get out of sync with revisions if a revision gets deleted, or
+ * if there's a bug. This function will update a document to match the most
+ * recent *version* in the revisions schema.
+ */
+export async function syncDocumentWithLatestRevision<T extends DbObject>(
+  collection: CollectionBase<T>,
+  document: T,
+  fieldName: string
+): Promise<void> {
+  const latestRevision = await Revisions.findOne(
+    {documentId: document._id,},
+    { sort: { version: -1 } }
+  )
+  if (!latestRevision) {
+    // Some documents are deletable, but typescript doesn't know that
+    if ((document as unknown as {deleted: boolean}).deleted) {
+      return
+    } else {
+      throw new Error(
+        `Document ${document._id} (${collection.collectionName}) has no revisions`
+      )
+    }
+  }
+  await collection.update(document._id, {
+    $set: {
+      [fieldName]: pick(latestRevision, revisionFieldsToCopy),
+      [`${fieldName}_latest`]: latestRevision._id
```

Comment:
This is going to also want to update (actually just unsetting is fine) the denormalized fields `contributionScores` and `htmlWithContributorAnnotations` (merged to devel just yesterday). I'll take care of this in another PR.